charts/merlin/templates/_helpers.tpl

```diff
@@ -85,9 +85,6 @@ app: {{ template "merlin.name" .}}
 release: {{ .Release.Name }}
 app.kubernetes.io/name: {{ template "merlin.name" . }}
 helm.sh/chart: {{ printf "%s-%s" .Chart.Name .Chart.Version | quote}}
-{{- if .Chart.AppVersion }}
```

Comment:
I believe we should ideally update the AppVersion for every breaking change in the chart, that makes it incompatible with older app versions.

Anyway, this label is set for several charts in this repo, and all caraml component charts except turing. Given this, I don't think we should remove it.

You've mentioned:

> misleading app.kubernetes.io/version=[app version] label for Merlin resources

Can you elaborate on what the confusion is?

Reply:
Hmm if we're using this `appVersion` label for more component charts than not (which I had originally thought was the case), then maybe we'd need to somehow refactor this value to make it reflective of the actual Merlin image tag value used.

I guess the PR description wasn't too clear on what the confusion was, but essentially, if the `appVersion` doesn't get actively updated, we'll start seeing inconsistencies with the Merlin image tag:
<img width="874" alt="Screenshot 2023-08-23 at 10 55 35 AM" src="https://github.com/caraml-dev/helm-charts/assets/36802364/01418105-5ca3-4c82-a424-cb613c073867">

Reply:
I see.. Here is an SO answer on how `appVersion` can be used: https://stackoverflow.com/a/60054111

Essentially, it need not correspond to the image version.

The fact that we don't always update the appVersion is another issue. 😅 Perhaps we can discuss this internally with @caraml-dev/platform on what the best way to use / update this value would be and adopt it in the future. For now, I think we can leave the label as is.

Reply:
Ahahha thanks for sharing that link!

> Essentially, it need not correspond to the image version.

Ah okay, yeah I guess it would've been clearer if we weren't versioning the app and image tags the same way, but yeah maybe we'll need more time to think about how to version these stuff consistently (or in a different way). I'll keep the label and merge the other changes then. Thanks a lot!